airbyte-integrations/connectors/source-s3/source_s3/v4/config.py

```diff
@@ -50,4 +56,10 @@ def validate_optional_args(cls, values):
             raise ValidationError(
                 "`aws_access_key_id` and `aws_secret_access_key` are both required to authenticate with AWS.", model=Config
             )
+
+        endpoint = values.get("endpoint")
+        if endpoint:
+            if endpoint.startswith("http://"):
+                raise ValidationError("The endpoint must be a secure HTTPS endpoint.", model=Config)
```

Comment:
As you allow endpoint without `https://` prefix I don't think this is enough to validate that the endpoint is secured. I would suggst making a `HEAD` request to the endpoint with and `https://` prefix and assert you get a 200 + a response with `https://` prefix.

Reply:
We ensure secure connections by configuring the boto3 client to use SSL and verify certificates, as shown in the code here: [boto3 SSL configuration](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-s3/source_s3/v4/stream_reader.py#L189-L194). This aligns with the boto3 documentation on enforcing SSL usage: [boto3 documentation](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/core/session.html#boto3.session.Session.client).

Reply:
Nice, and this client is used during `check` right?